sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -176,15 +190,13 @@ func GetRecoveryKind(err error) recoveryKind {
 	if errors.As(err, &rpcErr) {
 		code := rpcErr.RPCCode()
 
-		if code == 404 {
+		if code == RPCResponseCodeNotFound || code == RPCResponseCodeLockLost {
 			return RecoveryKindFatal
 		}
 
 		// this can happen when we're recovering the link - the client gets closed and the old link is still being
 		// used by this instance of the client. It needs to recover and attempt it again.
-		if code == 401 ||
-			// we lost the session lock, attempt link recovery
-			code == 410 {
+		if code == 401 {
```

Comment:
Add a named `const` for 401?

Reply:
Sounds good!

Reply:
I did some looking - these actually are supposed to map to HTTP status codes! I swapped out all of the constants for the http status code except for `RPCResponseCodeLockLost` since that one maps to StatusCodeGone and isn't necessarily intuitive like the others.